### PR TITLE
For #30158, path slash direction fix

### PR DESCRIPTION
**Pull request by jfboismenu-adsk**
Another fix for the path slash direction, this time for the path stored inside the configuration files.

### CODE REVIEW

File: tests/core_tests/test_api.py [hunk 1]
```diff
@@ -497,6 +497,37 @@ def test_with_mixed_slashes(self):
...
+        """
+        # only run this test on windows
+        if sys.platform == "win32":
+
+            get_install_locations_mock.return_value = {
+                # This path has multiple issues we've encountered in the wild
+                # It without any escaping sequence, it reads as
+                #   C:/configs\\site//project
+                # 1. it has whitespace at the begging and end.
+                # 2. Uses double slashes instead of single flash
```
Comment (jesseemond-adsk) [inline tests/core_tests/test_api.py]: "flash" is that a typo for "slash"?

Comment (jfboismenu-adsk) [inline tests/core_tests/test_api.py]: Yup! Thanks.

File: python/tank/pipelineconfig_utils.py [hunk 2]
```diff
@@ -159,8 +159,11 @@ def sanitize_path(path, separator=os.path.sep):
     """
     if path is None:
         return None
-    
-    # first, get rid of any slashes at the end
+
+    # ensure there is no white space around the path
+    path = path.strip()
```
Comment (manneohrstrom) [inline python/tank/pipelineconfig_utils.py]: maybe we could add an example in the doc string?

```
"   C:/configs\\\\site//project   "  - "c:\configs\site\project"
```

**Comment by manneohrstrom** [comment]
CR looks good. One super minor suggestion. Nice with the unit test! :)

**Pull request closed without merging**